File: find_npc_end.py
import json, os
os.environ.pop('AW_PYTHON_BINARY', None)
f = open(r"D:\pyproject\hero_workshop_react\src\data\npcs.ts", encoding="utf-8").read()
targets = ['changan_weizheng', 'changan_tangwang', 'changan_xuanzang']
for t in targets:
    idx = f.find("id: '" + t + "'")
    end = f.index("  },", idx + 100)
    segment = f[idx:end]
    last3 = segment.split("\n")[-3:]
    print(t, "lines:", len(segment.split("\n")), "last meaningful:", repr(last3[-2]) if len(last3)>1 else "")

File: find_npc_end.js
const fs = require('fs');
const f = fs.readFileSync('D:/pyproject/hero_workshop_react/src/data/npcs.ts', 'utf8');
const targets = ['changan_weizheng', 'changan_tangwang', 'changan_xuanzang'];

for (const n of targets) {
  const i = f.indexOf("id: '" + n + "'");
  if (i < 0) { console.log(n, 'NOT FOUND'); continue; }
  let d = 1, j = i;
  while (d > 0) d += (f[j] === '{') ? 1 : (f[j] === '}') ? -1 : 0, j++;
  const seg = f.substring(i, j);
  const lines = seg.split('\n');
  const last = lines[lines.length - 2]?.trim() ?? '';
  console.log(n, 'lines:', lines.length, 'last:', JSON.stringify(last).substring(0, 150));
}
